doniclient/tests/osc/test_hardware.py

```diff
@@ -126,19 +132,75 @@ class TestHardwareDelete(TestHardware):
     def setUp(self):
         super().setUp()
         self.cmd = hardware_cli.DeleteHardware(self.app, None)
+    
+    def test_hardware_delete_w_same_name(self):
+        # When delete when more than two resources exist with same name
+        arglist = [FAKE_HARDWARE_NAME]
+        verifylist = []
+        parsed_args = self.check_parser(self.cmd, arglist, verifylist)
+        # get will throw an Exception when used with name
+        self.hardware_mock.get.side_effect = Exception
+        self.hardware_mock.find.side_effect = Exception
+        # return two hardwares with same name
+        self.hardware_mock.list.return_value = [
+            hardware_fakes.FakeHardware.create_one_hardware(),
+            hardware_fakes.FakeHardware.create_one_hardware()
+        ]
+        try:
+            self.cmd.take_action(parsed_args)
+        except exceptions.CommandError as e:
+            self.assertEqual(e.args[0], f"More than one resource exists with the name or ID '{FAKE_HARDWARE_NAME}'.")
```

Comment:
Nice! I'm hoping we can reuse this general pattern to add better tests to Blazar as well.
👍